### ISSUE TITLE
Position file read error

### ISSUE THREAD

**Issue by speters** [jacmie/PosOrient#1]
Thx for this plugin to KiCAD.

But there seems to be something wrong when reading a POS file generated by my KiCAD (version info below):
![image](https://github.com/user-attachments/assets/b306f9ad-1a3e-43bc-81ba-234c037aba9b)

It looks like there is the references column missing and numerical data is off by 10x.

### Example **POS file** as generated by my KiCAD (via _File -> Fabrication Outputs -> Component Placement_)

```
### Footprint positions - created on 2024-10-18T18:42:23+0200 ###
### Printed by KiCad version 8.0.6
## Unit = mm, Angle = deg.
## Side : All
# Ref     Val       Package                                                                PosX       PosY       Rot  Side
C5                  C0805                                                               59.9990   -45.2667   90.0000  bottom
C7                  C_1206_3216Metric                                                   49.5360   -42.1933  180.0000  bottom
C8                  C0603                                                               31.8162   -28.7821    0.0000  bottom
C9                  C1206                                                               59.9990   -35.8764  -90.0000  bottom
C10                 C_1206_3216Metric                                                   59.9736   -29.5045  -90.0000  bottom
[...]
```

### Version info:

```
Application: KiCad x86_64 on x86_64

Version: 8.0.6, release build

Libraries:
	wxWidgets 3.2.6
	FreeType 2.13.3
	HarfBuzz 10.0.1
	FontConfig 2.15.0
	libcurl/8.10.1 OpenSSL/3.3.2 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libpsl/0.21.5 libssh2/1.11.0 nghttp2/1.63.0 nghttp3/1.6.0

Platform: Arch Linux, 64 bit, Little endian, wxGTK, X11, KDE, x11
OpenGL: Intel, Mesa Intel(R) HD Graphics 4000 (IVB GT2), 4.2 (Compatibility Profile) Mesa 24.2.4-arch1.1

Build Info:
	Date: Oct 16 2024 20:40:29
	wxWidgets: 3.2.6 (wchar_t,wx containers) GTK+ 3.24
	Boost: 1.86.0
	OCC: 7.8.1
	Curl: 8.10.1
	ngspice: 43
	Compiler: GCC 14.2.1 with C++ ABI 1019

Build settings:
	KICAD_USE_EGL=ON
```

**Comment by jacmie** [comment]
It's not a bug it's a feature ;) 
What I mean the KiCAD *.pos file format and the plugin's *.pos file format are different. Most information is the same, but KiCAD file contains # comments, additional side column and units. On the other hand the plugin's file has additional active flag column and is restricted to units: mm and deg.

I admit the same file extension might be confusing and I didn't expect that someone would want to open file in the plugin which wasn't saved by the plugin.

That being said what is the use case? Can you do your work by save and open with the plugin buttons?
I want to keep the plugin simple and doing it's thing, unless adding additional features (like import KiCAD file?) are really justified.

**Comment by speters** [comment]
I was searching for a placement script and came over yours and [kicad-parts-placer-pcm](https://github.com/TheJigsApp/kicad-parts-placer-pcm).

See the attached Screenshot for position x10 error:
![Screenshot_20241023-085232](https://github.com/user-attachments/assets/0fdadffd-02ba-4ac0-8278-0a18fd6b526a)

Beeing able to use a KiCAD produced POS file was just user expectancy. It is not an error, but it seems like a questionable decision to me to introduce a similar format when a native powerful format exists.

But I didn't just want to complain, I wanted to say thank you for your efforts and thx for making this plugin available. Thank you!

**Comment by jacmie** [comment]
I've checked the error and actually noticed few problems to fix there:
![image](https://github.com/user-attachments/assets/0c53d334-3719-4c96-8385-eb3e536a79c0)

- format values to maintain separation with white spaces
- X value in second row, which was edited, has comma not dot (need to check numbers <-> string conversion)
- if the list is long table after read does not expand

As to the file format I can't use the KiCAD *.pos due to the plugin's first column  "Active", which is missing there. 
I think I will just change plugin's file type extension to indicate this is something different. 

Thx for the feedback!

**Comment by jacmie** [comment]
I've fixed the issues and made new release v1.1.2 (not yet updated in the official KiCad repository).
@speters, could you make review does it work fine?

If you didn't install plugins manually:
Download release package from github on your local drive. 
Uninstall the plugin from the official KiCad repository.
In the Plugin and Content Manager click button Install from File ... and select the plugin zip package.

**Comment by speters** [comment]
Thx for working on this.

I tested your [v1.1.2 release](https://github.com/jacmie/PosOrient/releases/tag/v1.1.2).

1. opened PCB (in this case [SailorHat PCB](https://github.com/hatlabs/SH-RPi-hardware))
2. click PosOrient
![Screenshot_20241023-085232](https://github.com/user-attachments/assets/01205f57-6653-4e8a-a0b8-246c1d04402c)
4. "Update" list and save "cpf" file
5. show Footprint Properties to compare position
![Screenshot_20241030-172141](https://github.com/user-attachments/assets/298b4ff4-fca5-466a-a9f0-bfbf663d0e44)
6. move some footprints
7. open PosOrient and load "cpf" file saved in #3 and click "Orient"
![Screenshot_20241030-172322](https://github.com/user-attachments/assets/1b5364bf-a91f-4dc6-94c7-c1417f204906)
8. observe positions: positioning was OK

Note the position values having changed. Also note that position does not match position shown in Footprint Properties.
This seems bit weird to me, maybe different coordinate systems and/or different units are used.

**Comment by jacmie** [comment]
I've cloned the [SailorHat PCB](https://github.com/hatlabs/SH-RPi-hardware) case and made checks. Noticed two problems:
- few footprint names were too long
- few footprint names had spaces and weren't read back
![Screenshot from 2024-11-01 23-49-08](https://github.com/user-attachments/assets/034c2c24-e723-4b00-a2aa-5637c91dfbcc)
![Screenshot from 2024-11-01 23-51-36](https://github.com/user-attachments/assets/dec136d4-f71a-42ee-b93a-36e292045a2e)

I haven't notice differences of position values in the plugin, saved file vs footprint properties. 

I've fixed it all and added some warnings about possibly corrupted data.
Now the plugin seems to work fine with the SailorHat case.
Check the release v1.1.3
